public/js/draw/draw.js

```diff
@@ -38,47 +38,127 @@ function getPosition(elem) {
         var node = document.createElementNS(xmlns, 'rect');
+        node.setAttribute('rx', 4);
+        node.setAttribute('ry', 4);
+    }
+
+        var g = document.createElementNS(xmlns, 'g');
 
-        g.setAttribute('class', 'node');
+        g.setAttribute('class', gClass);
         g.setAttribute('id', 'g' + nodeId);
         g.setAttribute('data-active', 'active');
-        g.setAttribute('data-group', nodeColourId);
 
-        node.setAttribute('x', position.x);
-        node.setAttribute('y', position.y);
-        node.setAttribute('rx', 4);
-        node.setAttribute('ry', 4);
-        node.setAttribute('id', 'n' + nodeId);
+        node.setAttribute('x', posX);
+        node.setAttribute('y', posY);
+        node.setAttribute('fill', nodeFill);
+        node.setAttribute('id', nodeId_);
         node.setAttribute('width', nodeWidth);
         node.setAttribute('height', nodeHeight);
-        node.setAttribute('class', 'node');
-        node.parents = [];
-        node.kids = [];
+        node.setAttribute('tolerance', nodeTolerance);
+        node.predecessors = [];
+        node.successors = [];
         // note: children doesn't work because javascript objects already have a children attribute
         node.inEdges = [];
         node.outEdges = [];
 
         g.appendChild(node);
         svgDoc.appendChild(g);
-        document.getElementById('n' + nodeId).addEventListener('mousedown', nodeClicked, false);
-
-        select(document.getElementById('n' + nodeId));
+        document.getElementById(nodeId_).addEventListener('mousedown', nodeClicked, false);
+
+        // Input Text into Shape
+        if (jsonObj){
+            for (var i = 0; i < jsonObj.text.length; i++) {
+                textPosX = jsonObj.pos[0];
+                if (jsonObj.type_ !== 'BoolNode') {
+                    textPosX += (jsonObj.width/2)
+                }
 
+                textPosY = jsonObj.text[i].pos[1] - (jsonObj.height/4);
+                textStr = jsonObj.text[i].text || '';
+                textFill = jsonObj.text[i].fill;
+                textAlign = jsonObj.text[i].align;
+                textId = 't' + jsonObj.id_;
+
+                var code = document.createElementNS(xmlns, 'text');
+                code.setAttributeNS(null, 'id', textId);
+                code.setAttributeNS(null, 'fill', textFill);
+                code.setAttributeNS(null, 'align', textAlign);
+                code.setAttributeNS(null, 'x', textPosX);
+                code.setAttributeNS(null, 'y', textPosY);
+                code.setAttributeNS(null, 'class', 'mylabel'); // note: label is a class in bootstrap
+                var textNode = document.createTextNode(textStr);
+                code.appendChild(textNode);
+                g.appendChild(code);
+                document.getElementById(textId).addEventListener('mousedown', nodeClicked, false);
+            };
+        }
+        select(document.getElementById(nodeId_));
         nodeId += 1;
```

Comment:
1. I'm not sure what this line does if it's at the end of the function.
2. Are you missing a `}`? The indentation seems off.